openapi_python_client/parser/properties/__init__.py

```diff
@@ -388,14 +405,15 @@ def _property_from_data(
     """ Generate a Property from the OpenAPI dictionary representation of it """
     name = utils.remove_string_escapes(name)
     if isinstance(data, oai.Reference):
-        reference = Reference.from_ref(data.ref)
-        existing = schemas.enums.get(reference.class_name) or schemas.models.get(reference.class_name)
-        if existing:
-            return (
-                attr.evolve(existing, required=required, name=name),
-                schemas,
+        return _property_from_ref(name=name, required=required, nullable=False, data=data, schemas=schemas)
+
+    for attribute in ["allOf", "anyOf", "oneOf"]:
+        sub_data = getattr(data, attribute)
+        if sub_data and len(sub_data) == 1 and isinstance(sub_data[0], oai.Reference):
+            return _property_from_ref(
+                name=name, required=required, nullable=data.nullable, data=sub_data[0], schemas=schemas
             )
```

Comment:
I don't think there's anything illegal about having more than one of those properties, so we only want to do the pass through if only 1 ref exists in all of those properties. So something like:

```suggestion
    sub_data = (data.allOf or []) + data.anyOf + data.oneOf
    if len(sub_data) == 1 and isinstance(sub_data[0], oai.Reference):
        return _property_from_ref(
            name=name, required=required, nullable=data.nullable, data=sub_data[0], schemas=schemas
        )
```
This also means no `getattr` so we get better type checking!